src/Umbraco.Deploy.Contrib.Connectors/ValueConnectors/BlockEditorValueConnector.cs

```diff
@@ -0,0 +1,245 @@
+                return null;
+
+            if (svalue.DetectIsJson() == false)
+                return null;
+            var blockEditorValue = JsonConvert.DeserializeObject<BlockEditorValue>(svalue);
+
+            if (blockEditorValue == null)
+                return null;
+
+            // get all the content types used in block editor items
+            var allContentTypes = blockEditorValue.Data.Select(x => x.ContentTypeKey)
+                .Distinct()
+                .ToDictionary(a => a, a =>
+                {
+                    if (!Guid.TryParse(a, out var keyAsGuid))
+                        throw new InvalidOperationException($"Could not parse ContentTypeKey as GUID {keyAsGuid}.");
+                    return _contentTypeService.Get(keyAsGuid);
+                });
+
+            //Ensure all of these content types are found
+            if (allContentTypes.Values.Any(contentType => contentType == null))
+            {
+                throw new InvalidOperationException($"Could not resolve these content types for the Block Editor property: {string.Join(",", allContentTypes.Where(x => x.Value == null).Select(x => x.Key))}");
+            }
+
+            //Ensure that these content types have dependencies added
+            foreach (var contentType in allContentTypes.Values)
+            {
+                dependencies.Add(new ArtifactDependency(contentType.GetUdi(), false, ArtifactDependencyMode.Match));
+            }
+
+            foreach (var block in blockEditorValue.Data)
+            {
+                var contentType = allContentTypes[block.ContentTypeKey];
+
+                foreach (var key in block.PropertyValues.Keys.ToArray())
+                {
+                    var propType = contentType.CompositionPropertyTypes.FirstOrDefault(x => x.Alias == key);
+
+                    if (propType == null)
+                    {
+                        _logger.Debug<NestedContentValueConnector>($"No property type found with alias {key} on content type {contentType.Alias}");
+                        continue;
+                    }
+
+                    // fetch the right value connector from the collection of connectors, intended for use with this property type.
+                    // throws if not found - no need for a null check
+                    var propValueConnector = ValueConnectors.Get(propType);
+
+                    // pass the value, property type and the dependencies collection to the connector to get a "artifact" value
+                    var val = block.PropertyValues[key];
+                    object parsedValue = propValueConnector.ToArtifact(val, propType, dependencies);
+
+                    // getting Map image value umb://media/43e7401fb3cd48ceaa421df511ec703c to (nothing) - why?!
+                    _logger.Debug<BlockEditorValueConnector>("Map " + key + " value '" + block.PropertyValues[key] + "' to '" + parsedValue + "' using " + propValueConnector.GetType() + " for " + propType);
```

Comment:
Please use string interpolation instead please `$"Map {key} {value} ....."`
Also please use the nice Serilogger to pass the values into something that's easier to search/find